browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -434,23 +434,18 @@ void RewardsInternalsDOMHandler::GetAdDiagnostics(
                      weak_ptr_factory_.GetWeakPtr()));
 }
 
-void RewardsInternalsDOMHandler::OnGetAdDiagnostics(const bool success,
-                                                    const std::string& json) {
+void RewardsInternalsDOMHandler::OnGetAdDiagnostics(
+    absl::optional<base::Value::List> diagnostics) {
+  DCHECK(diagnostics);
+  DCHECK(diagnostics.has_value());
```

Comment:
```c++
  DCHECK(diagnostics);
  DCHECK(diagnostics.has_value());
```
These two checks are doing the same thing. The first one is preferable, so you can keep only that one.

Question though: unless the user of optional in this argument is part of some type signature in mojom, wouldn't it be possible to pass this list by value? That would entirely avoid the need of the check and the possibility of invalid state.

Reply:
Fixed DCHECK

Reply:
Good catch! Changed `DCHECK` as we should fail gracefully for `absl::nullopt` if bat ads is not bound